### ISSUE TITLE
TDigest.Compress bug?

### ISSUE THREAD

**Issue by dmitry-si** [caio/go-tdigest#9]
```
t.Summary = newSummary(estimateCapacity(t.Compression), oldTree.Resolution)

# missing
t.Count = 0
```

t.Add also does t.Count += count, so t.Count is doubled on each TDigest.Compress

**Comment by caio** [comment]
Hi!

This is now fixed after merging #10 right?

on `Add()`, this will trigger setting the count to the correct value:
```
if t.summary.Len() == 0 {
    t.summary.Add(value, count)
    t.count = count
    return nil
}
```

Would it make things more understandable if on `Compress` we explicitly set it to zero?

Nice catch!

**Comment by caio** [comment]
I've added a `count = 0` on cc858f8fe45084d0b3f988461563ea59579e8e12 and I'm closing this issue as I believe the issue is fixed. Feel free to reopen if I'm mistaken.
